apollo-federation/src/operation/mod.rs

```diff
@@ -104,6 +104,98 @@ fn same_value(left: &executable::Value, right: &executable::Value) -> bool {
     }
 }
 
+/// Sort an input value, which means specifically sorting their object values by keys (assuming no
+/// duplicates). This is used for hashing input values in a way consistent with [same_value()].
+fn sort_value(value: &executable::Value) -> executable::Value {
```

Comment:
The `apollo-compiler` API is really made for mutation rather than creating new instances. All the `sort_*` functions should ideally take a `&mut executable::WhateverTheType` and mutate. It's also easier to implement them that way :)

Reply:
Makes sense! Pushed up changes to switch to `&mut` in the latest commit.

Reply:
Alright, pushed up with merged conflicts resolved.